pkg/deploy/controller/imagechange/controller.go

```diff
@@ -111,10 +111,10 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
 	}
 
 	if anyFailed {
-		return fatalError(fmt.Sprintf("couldn't update some deployment configs for trigger on image stream %q", imageapi.LabelForStream(stream)))
```

Comment:
Since we're processing image stream changes, we can't really avoid
recalculating all of them anyway (otherwise we'd have to carry state).  So
the only possible error that would terminate would be "did all of them fail
with not-found" which seems unlikely.

On Sun, Jul 10, 2016 at 10:59 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Actually, since it's a whole group, it's a lot harder to calculate the
> diffs.  I don't see any reason not to retry on all other errors, since
> they'll either be no-op, or potentially continue.  The case I can imagine
> would be a totally unexpected failure (like the deployment is in a
> permanently screwed up state), but those we'd need to fix as well.
>
> On Sun, Jul 10, 2016 at 10:56 PM, Jordan Liggitt <notifications@github.com
> > wrote:
>
>> In pkg/deploy/controller/imagechange/controller.go
>> <https://github.com/openshift/origin/pull/9578#discussion_r70196827>:
>>
>> > @@ -111,10 +111,10 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
>> >  	}
>> >
>> >  	if anyFailed {
>> > -		return fatalError(fmt.Sprintf("couldn't update some deployment configs for trigger on image stream %q", imageapi.LabelForStream(stream)))
>>
>> yeah, or just retry specifically on conflict errors
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/openshift/origin/pull/9578/files/a850e5e780643ff932482d6a9224cbfbb137f332..ddd80e057bc3163119b64497ec1f0e1ba8b532fb#r70196827>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ABG_p_gQo3zjUcamv_jYHDF8LXPK8fFoks5qUbDcgaJpZM4I-xTE>
>> .
>>
>
>